include/CL/opencl.hpp

```diff
@@ -10253,6 +10342,530 @@ namespace compatibility {
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_FILL_BUFFER_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
+    cl_int commandFillImage(const CommandQueue* command_queue,
+        const Image& image,
+        cl_float4 fillColor,
+        const array<size_type, 3>& origin,
+        const array<size_type, 3>& region,
+        const vector<cl_sync_point_khr>* sync_points_vec = NULL,
+        cl_sync_point_khr* sync_point = NULL,
+        cl_mutable_command_khr* mutable_handle = NULL)
+    {
+        cl_sync_point_khr tmp_sync_point;
+        cl_int error = detail::errHandler(
+            pfn_clCommandFillImageKHR(object_,
+                (command_queue != NULL) ? command_queue->operator()() : NULL,
+                image(),
+                static_cast<void*>(&fillColor),
+                origin.data(),
+                region.data(),
+                (sync_points_vec != NULL) ? (cl_uint) sync_points_vec->size() : 0,
+                (sync_points_vec != NULL && sync_points_vec->size() > 0) ? (cl_sync_point_khr*) &sync_points_vec->front() : NULL,
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_FILL_IMAGE_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
+    cl_int commandNDRangeKernel(const CommandQueue* command_queue,
+        const cl_ndrange_kernel_command_properties_khr* properties,
+        const Kernel& kernel,
+        const NDRange& offset,
+        const NDRange& global,
+        const NDRange& local = NullRange,
+        const vector<cl_sync_point_khr>* sync_points_vec = NULL,
+        cl_sync_point_khr* sync_point = NULL,
+        cl_mutable_command_khr* mutable_handle = NULL)
+    {
+        cl_sync_point_khr tmp_sync_point;
+        cl_int error = detail::errHandler(
+            pfn_clCommandNDRangeKernelKHR(object_,
+                (command_queue != NULL) ? command_queue->operator()() : NULL,
+                properties,
+                kernel(),
+                (cl_uint) global.dimensions(),
+                offset.dimensions() != 0 ? (const size_type*) offset : NULL,
+                (const size_type*) global,
+                local.dimensions() != 0 ? (const size_type*) local : NULL,
+                (sync_points_vec != NULL) ? (cl_uint) sync_points_vec->size() : 0,
+                (sync_points_vec != NULL && sync_points_vec->size() > 0) ? (cl_sync_point_khr*) &sync_points_vec->front() : NULL,
+                (sync_point != NULL) ? &tmp_sync_point : NULL,
+                mutable_handle),
+            __COMMAND_NDRANGE_KERNEL_KHR_ERR);
+
+        if (sync_point != NULL && error == CL_SUCCESS)
+            *sync_point = tmp_sync_point;
+
+        return error;
+    }
+
```

Comment:
there are two more functions that are related to command buffer (mutable dispatch) that we can support in c++ binding : 
clUpdateMutableCommandsKHR() , clGetMutableCommandInfoKHR()

Reply:
OK Ben. For clGetMutableCommandInfoKHR best will be to create separately class MutableCommandKhr. Do you think is a proper way ?

Reply:
I think that no need for class for MutableCommandKhr , you can just implement this like all the functions in commandBuffer class right?

Reply:
I think it would be better to have this as separated class, but maybe I am wrong.
Please check my following commit.